kaska/interp_fix.py

```diff
@@ -0,0 +1,30 @@
+import numpy as np
+from numba import jit
+
+'''
+This is a linear interpolation over 1 axis
+using numpy.interp function. It uses numba 
+to speed up filling gaps in the for loop.
+
+Feng Yin
+Department of Geography, UCL
+ucfafyi@ucl.ac.uk
+LICENSE: GNU GENERAL PUBLIC LICENSE V3 
+'''
+
+@jit(nopython=True)
+def interp1d(newx, oldx, oldy):
+    if oldx.shape[0] != oldy.shape[0]:
+        raise ValueError('oldx and oldy must have the same shape in first axis.')
+    oldy_shape = oldy.shape[1:]
+    new_shape  = newx.shape[0]
+    oldy_t = oldy.reshape(oldx.shape[0], -1).T
+    pix_num = oldy_t.shape[0]
+    newy = np.zeros((pix_num, new_shape))*np.nan
+    for i in range(pix_num):
```

Comment:
```python
for i, y in enumerate(oldy_t):
```

Reply:
Also in master. Will fix there and rebase/merge/cherrypick.

Reply:
This causes an internal numba error in the unit test. I don't think the numba implementation can cope with the enumerate() call.

Reply:
Interesting! Maybe is to do with the numba version...  Their [docs](https://numba.pydata.org/numba-doc/dev/reference/pysupported.html) say it can... well, let's leave that. Could you write a comment there so we don't try to do it in the future?